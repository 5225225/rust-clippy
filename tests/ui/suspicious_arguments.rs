// edition:2015
#![warn(clippy::suspicious_arguments)]
#![allow(anonymous_parameters)]

trait AnonymousArgs {
    fn scale(&self, usize, usize) {}
}

fn scale_this<A: AnonymousArgs>(value: A) {
    // We don't expect a lint, we just don't want an ICE here.
    let width = 1;
    let height = 2;
    value.scale(width, height);
    A::scale(&value, width, height);
}

fn resize(width: usize, height: usize) {}

struct Bitmap;

impl Bitmap {
    fn new(width: usize, height: usize) -> Self {
        Bitmap
    }
}

fn main() {
    let width = 0;
    let height = 0;

    resize(height, width);
    Bitmap::new(height, width);

    resize(0, width);
    resize(height, 0);
    resize(height, height);
    resize(width, width);

    let f = vec![0_u8];
    let iterable = |_| { true };

    itertools::all(f, iterable);

    let mut xvalue = 42;
    let mut yvalue = 42;
    let x = &mut xvalue;
    let y = &mut yvalue;

    std::mem::swap(y, x);
    
}
