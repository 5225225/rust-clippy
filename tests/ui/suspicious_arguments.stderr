error: these arguments are possibly swapped
  --> $DIR/suspicious_arguments.rs:31:12
   |
LL |     resize(height, width);
   |            ^^^^^^  ^^^^^
   |
   = note: `-D clippy::suspicious-arguments` implied by `-D warnings`
note: the arguments are defined here
  --> $DIR/suspicious_arguments.rs:17:11
   |
LL | fn resize(width: usize, height: usize) {}
   |           ^^^^^         ^^^^^^

error: these arguments are possibly swapped
  --> $DIR/suspicious_arguments.rs:32:17
   |
LL |     Bitmap::new(height, width);
   |                 ^^^^^^  ^^^^^
   |
note: the arguments are defined here
  --> $DIR/suspicious_arguments.rs:22:12
   |
LL |     fn new(width: usize, height: usize) -> Self {
   |            ^^^^^         ^^^^^^

error: these arguments are possibly swapped
  --> $DIR/suspicious_arguments.rs:42:20
   |
LL |     itertools::all(f, iterable);
   |                    ^  ^^^^^^^^
   |
note: the arguments are defined here
  --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/itertools-0.10.1/src/free.rs:151:18
   |
LL | pub fn all<I, F>(iterable: I, f: F) -> bool
   |                  ^^^^^^^^     ^

error: these arguments are possibly swapped
  --> $DIR/suspicious_arguments.rs:49:20
   |
LL |     std::mem::swap(y, x);
   |                    ^  ^
   |
note: the arguments are defined here
  --> /home/jess/.rustup/toolchains/nightly-2022-07-28-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/mem/mod.rs:711:22
   |
LL | pub const fn swap<T>(x: &mut T, y: &mut T) {
   |                      ^          ^

error: aborting due to 4 previous errors

